lambdas/backend/express-route-handlers.js

```diff
@@ -279,6 +288,136 @@ function postFeedback(req, res) {
     }
 }
 
+async function getAdminCatalogVisibility(req, res) {
+    console.log(`GET /admin/catalog/visibility for Cognito ID: ${req.apiGateway.event.requestContext.identity.cognitoIdentityId}`)
+    try {
+
+        let visibility = { apiGateway: [] }, catalogObject = await catalog(),
+            apis = (await exports.apigateway.getRestApis().promise()).items
+
+        console.log(`network request: ${JSON.stringify(await exports.apigateway.getRestApis().promise(), null, 4)}`)
+
+        console.log(`apis: ${apis}`)
+
+        let promises = []
+        apis.forEach((api) => {
+            promises.push(
+                exports.apigateway.getStages({ restApiId: api.id }).promise()
+                    .then((response) => response.item)
+                    .then((stages) => stages.forEach(stage => visibility.apiGateway.push({
+                        id: api.id,
+                        stage: stage.stageName,
+                        visibility: false
+                    })))
+            )
+        })
+        await Promise.all(promises)
```

Comment:
Is it worth it to catch failures on each `getStage` call? So that if one fails, we at least return the rest?

Reply:
Probably not. I'd rather not have a UI than have a UI with incorrect information?

Reply:
As a user, would it be preferable to have an all-or-none experience?